### PR TITLE
JsonIgnore ClientLogger

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR adds `@JsonIgnore` to `ClientLogger` fields on objects that will be serialized.

### CODE REVIEW

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/SentimentSkill.java [hunk 1]
```diff
@@ -24,6 +25,7 @@
 @JsonTypeName("#Microsoft.Skills.Text.SentimentSkill")
 @Fluent
 public final class SentimentSkill extends SearchIndexerSkill {
+    @JsonIgnore
```
Comment (pallavit) [inline sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/SentimentSkill.java:28]: This is a good one! How did you find this? Should we add this in our code quality checks - not have ClientLogger w/o JsonIgnore in our models.

**Review by pallavit — APPROVED** [review]
:shipit:

File: sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/SentimentSkill.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/SentimentSkill.java:28]: Tests against a live Search service failed 😞

**Pull request merged**